### PR TITLE
PUBDEV-8561: h2o.upload_mojo should also work for POJO

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-algos/src/main/java/hex/generic/PojoLoader.java [hunk 1]
```diff
@@ -18,23 +18,35 @@ class PojoLoader {
...
-            String path = URI.create(pojoKey.toString()).getPath();
-            String fileName = new File(path).getName();
-            if (fileName.endsWith(POJO_EXT)) {
-                fileName = fileName.substring(0, fileName.length() - POJO_EXT.length());
-            }
-            return compileAndInstantiate(fileName, pojoCode);
+            className = inferClassName(pojoKey, modelId);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to automatically infer POJO class name, " +
+                    "please specify the POJO class name explicitly be setting `model_id` parameter.");
```
Comment (koniecsveta) [inline h2o-algos/src/main/java/hex/generic/PojoLoader.java:28]: typo be -> by

File: h2o-algos/src/main/java/hex/generic/PojoLoader.java [hunk 2]
```diff
@@ -18,23 +18,35 @@ class PojoLoader {
 
     private static final String POJO_EXT = ".java";
 
-    static GenModel loadPojoFromSourceCode(ByteVec sourceVec, Key<Frame> pojoKey) throws IOException {
-        String pojoCode = IOUtils.toString(sourceVec.openStream());
+    static GenModel loadPojoFromSourceCode(ByteVec sourceVec, Key<Frame> pojoKey, String modelId) throws IOException {
+        final String pojoCode = IOUtils.toString(sourceVec.openStream());
```
Comment (koniecsveta) [inline h2o-algos/src/main/java/hex/generic/PojoLoader.java:22]: is the stream being closed?

File: h2o-algos/src/main/java/hex/generic/PojoLoader.java (hunk 1 above)
Comment (koniecsveta) [inline h2o-algos/src/main/java/hex/generic/PojoLoader.java:28]: At first sight, this exception message confuses me. It seems to me, that if there happens an exception during inferClassName() commands, it doesn't matter whether modelId is set to something or whether it is a null.

Comment (michalkurka) [reply h2o-algos/src/main/java/hex/generic/PojoLoader.java:28]: you are right, this doesn't do anything!

Comment (michalkurka) [reply h2o-algos/src/main/java/hex/generic/PojoLoader.java]: fixed, thank you for pointing it out

**Pull request merged**